docs/spec/sources.md

```diff
@@ -362,5 +375,38 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects). It
+is highly-recommended to produce the `knsource` CloudEvent extension.
```

Comment:
```suggestion
identifying the originating [Source Custom Object](#source-custom-objects). It
is highly-recommended to produce the `knsource` CloudEvent extension because without one, users might not be able to filter events coming from different custom objects that belong to the same Source CRD. See the Examples section below.
```
Just another suggestion.

Reply:
That's what I also suggested, clarify the _why_ in the same sentence.

A correction though: objects do not "belong" to a CRD. If we use your phrasing, it should read "objects of a certain source kind".

Reply:
I feel this does not belong in the spec, which is meant to be concise and prescriptive. I would rather keep the use case example in a separate paragraph.

Reply:
@lionelvillard fair enough, what do you think of this version of wording?